skbio/metadata/_interval.py

```diff
@@ -0,0 +1,334 @@
+
+        invs = set()
+
+        # Find queries by interval
+        if intervals is not None:
+            for value in intervals:
+                invs.update(self._query_interval(value))
+
+        # Find queries by feature attribute
+        if len(invs) == 0 and metadata is not None:
+            invs = set(self._metadata)
+
+        if metadata is not None:
+            invs = self._query_attribute(invs, metadata)
+        return list(invs)
+
+    def drop(self, intervals=None, boundaries=None, metadata=None):
+        """ Drops Interval objects according to a specified query.
+
+        Drops all Interval objects that matches the query.
+
+        Parameters
+        ----------
+        intervals : iterable of tuple of ints
+            A list of intervals associated with the Interval object.
+        boundaries : iterable of tuple of bool
+            A list of boundaries associated with the Interval object.
+        metadata : dict
+            A dictionary of key word attributes associated with the
+            Interval object.
+
+        Examples
+        --------
+        >>> from skbio.metadata import IntervalMetadata
+        >>> interval_metadata = IntervalMetadata()
+        >>> interval_metadata.add(intervals=[(0, 2), (4, 7)],
+        ...                       boundaries=None, metadata={'name': 'sagA'})
+        >>> interval_metadata.add(intervals=[(40, 70)],
+        ...                       boundaries=None, metadata={'name': 'sagB'})
+        >>> interval_metadata.drop(metadata={'name': 'sagA'})
+        >>> interval_metadata.query(metadata={'name': 'sagA'})
+        []
+        """
+        if intervals is None:
+            intervals = []
+        if metadata is None:
+            metadata = {}
+
+        queried_invs = self.query(intervals=intervals,
+                                  boundaries=boundaries,
+                                  metadata=metadata)
+        new_invs = []
+        # iterate through queries and drop them
+        for inv in self._metadata:
+            if inv not in queried_invs:
+                new_invs.append(inv)
+        self._metadata = new_invs
+        self._is_stale_tree = True
+
+    def __eq__(self, other):
+        return sorted(self._metadata) == sorted(other._metadata)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __repr__(self):
+        if len(self._metadata) < 10:
+            return str(self._metadata)
+        else:
+            return str(self._metadata[:5] + ['...'] + self._metadata[-5:])
+
+
+def _polish_interval(interval):
+    if isinstance(interval, tuple):
+        if len(interval) == 0:
+            return None
+        if len(interval) == 2:
+            start, end = interval
+        else:
+            raise ValueError("`start` and `end` aren't correctly specified")
+    else:
+        raise ValueError('The args must be associated with '
+                         'a tuple when querying')
+    return start, end
```

Comment:
There doesn't seem to be a reason to return `start, end` because that's just the unpacked `interval` and you're not updating those values in this function. Maybe clearer to rename to something like `_assert_valid_interval`? We've used `_assert_valid_*` methods in other parts of the codebase so would be nice to be consistent.